### PR TITLE
[3.3.5] Fix instance trash spawn behavior

### PR DESCRIPTION
**Pull request by Treeston**
Implement new CONDITION_SOURCE_TYPE_CREATURE, which allows the condition system to prevent spawning of creatures.

The immediate use is that this allows us to have blizzlike trash respawn behavior in instances, with trash no longer respawning after the associated boss is dead, and also despawning on soft reset if the boss is dead (issue #15682).

If a condition for checking instance group's LFD id is implemented, this system could also be used to spawn holiday bosses only when queued for them using LFD (see discussion in PR #14884).

As an example, the included sql update file adds conditions to all appropriate creatures in Naxxramas, causing them to have correct respawn behavior.

### CODE REVIEW

File: src/server/game/Entities/Creature/Creature.cpp [hunk 1]
```diff
@@ -222,7 +222,11 @@ void Creature::AddToWorld()
 
         GetMap()->GetObjectsStore().Insert<Creature>(GetGUID(), this);
         if (m_spawnId)
+        {
             GetMap()->GetCreatureBySpawnIdStore().insert(std::make_pair(m_spawnId, this));
+            if (!sConditionMgr->IsObjectMeetingNotGroupedConditions(CONDITION_SOURCE_TYPE_CREATURE, m_spawnId, this))
+                AddObjectToRemoveList();
```
Comment (Shauren) [inline src/server/game/Entities/Creature/Creature.cpp]: not approving this hack

**Comment by Aokromes** [comment]
This branch has conflicts that must be resolved
 Use the command line to resolve conflicts before continuing.

**Comment by Treeston** [comment]
Like I said a few times already - this won't be merged.
It'll be part of dynamic spawns once that gets merged, and @r00ty-tc and myself will find a less hacky way of implementing it.

**Comment by r00ty-tc** [comment]
Pretty sure the version already in dynspawn branch stops the removal after spawning. They're simply not spawned at all if they don't meet the condition.

**Comment by Treeston** [comment]
Still hacky with creating the dummy worldobject.

**Pull request closed without merging**